### PR TITLE
[10.0][ADD] sale_invoice_policy

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Cedric-Pigeon @jbeficent @acsonefho

**Comment by rousseldenis** [comment]
WIP

File: sale_invoice_policy/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        """
+        Apply the invoice_policy given by context (if exist) otherwise use the
+        default invoice policy given by the field with this same name.
+        If the product is type = 'service', we don't have to apply the invoice
+        policy given by the context.
+        :return:
+        """
+        invoice_policy = self.env.context.get('invoice_policy')
+        for tmpl in self:
+            if tmpl.type != 'service' and invoice_policy:
```
Comment (AaronHForgeFlow) [inline sale_invoice_policy/models/product_template.py:46]: It's possible to set a delivered qty in the sale order lines for services so an invoice policy based on delivered quantities will still make sense to me.

Comment (rousseldenis) [reply sale_invoice_policy/models/product_template.py:46]: @hveficent Still in WIP. I'm migrating this from one of our repo. Still adapting

**Comment by rafaelbn** [comment]
Hi,

When I try to add a line in the sale order I get error:

`Error: Unknown field is_template in domain [["is_template","=",true]]`
```
http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:1767
Traceback:
compute_domain@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:1767:42
compute_domain@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:2188:339
init/<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:2222:341
trigger@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:3104:166
trigger@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:3108:115
on_form_changed@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:2138:93
load_record/<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:2135:1
then/</</<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:547:678
fire@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:541:281
add@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:542:467
then/</<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:547:631
each@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:370:758
then/<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:547:553
Deferred@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:548:189
then@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:547:518
load_record@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:2134:382
trigger@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:3104:166
trigger@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:3108:115
load_defaults/<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/549-bde8409/web.assets_backend.js:2137:198
then/</</<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:547:678
fire@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:541:281
fireWith@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:546:198
Deferred/</deferred[tuple[0]]@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:548:31
fire@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:541:281
fireWith@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:546:198
then/</</<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:547:849
fire@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:541:281
fireWith@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:546:198
then/</</<@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:547:849
fire@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:541:281
fireWith@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:546:198
done@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:937:86
callback@http://3325251-642-ed7327.runbot1.odoo-community.org/web/content/543-ae44fa4/web.assets_common.js:957:15
```

**Comment by rousseldenis** [comment]
@rafaelbn Needs review.
These stack trace seems to come from sale_generator module.

**Comment by rousseldenis** [comment]
@rafaelbn Made an issue https://github.com/OCA/sale-workflow/issues/644

**Review by acsonefho — APPROVED** [review]
LGTM

**Comment by rousseldenis** [comment]
@OCA/crm-sales-marketing-maintainers

**Pull request merged**